descqa/ColorDistribution.py

```diff
@@ -179,73 +192,93 @@ def run_on_single_catalog(self, catalog_instance, catalog_name, output_dir):
+            if self.obs_r_mag_limit:
+                f.write('r_mag < %2.3f\n\n'%(self.obs_r_mag_limit))
+            elif self.Mag_r_limit:
+                f.write('Mag_r < %2.3f\n\n'%(self.Mag_r_limit))
             for color in self.colors:
-                if not ((color in self.obs_color_dist) and (color in mock_color_dist)):
+                if self.validation_catalog and not ((color in self.obs_color_dist) and (color in mock_color_dist)):
                     continue
-                f.write("Median "+color+" difference (obs - mock) = %2.3f\n"%(color_shift[color]))
-                f.write(color+": {} = {:2.6f}\n".format('CvM statistic', cvm_omega[color]))
-                f.write(color+" (shifted): {} = {:2.6f}\n".format('CvM statistic', cvm_omega_shift[color]))
-                f.write("\n")
+                if self.validation_catalog:
+                    f.write("Median "+color+" difference (obs - mock) = %2.3f\n"%(color_shift[color]))
+                    f.write(color+": {} = {:2.6f}\n".format('CvM statistic', cvm_omega[color]))
+                    f.write(color+" (shifted): {} = {:2.6f}\n".format('CvM statistic', cvm_omega_shift[color]))
+                    f.write("\n")
 
         return TestResult(inspect_only=True)
 
 
     def make_plots(self, mock_color_dist, color_shift, cvm_omega, cvm_omega_shift, catalog_name, output_dir):
-        nrows = int(np.ceil(len(self.colors)/2.))
+        available_colors = [c for c in self.colors if c in mock_color_dist]
+        print(available_colors)
+        nrows = int(np.ceil(len(available_colors)/2.))
         fig_pdf, axes_pdf = plt.subplots(nrows, 2, figsize=(8, 3.5*nrows))
         fig_cdf, axes_cdf = plt.subplots(nrows, 2, figsize=(8, 3.5*nrows))
+        title = ''
+        if self.obs_r_mag_limit:
+            title = '$m_r < {:2.1f},  {:.1f} < z < {:.1f}$'.format(self.obs_r_mag_limit, self.zlo, self.zhi)
+        elif self.Mag_r_limit:
+            title = '$M_r < {:2.1f},  {:.1f} < z < {:.1f}$'.format(self.Mag_r_limit, self.zlo, self.zhi)
 
-        for ax_cdf, ax_pdf, color in zip(axes_cdf.flat, axes_pdf.flat, self.colors):
+        print(mock_color_dist.keys())
+        for ax_cdf, ax_pdf, color in zip(axes_cdf.flat, axes_pdf.flat, available_colors):
 
-            if not ((color in self.obs_color_dist) and (color in mock_color_dist)):
+            if self.validation_catalog and not ((color in self.obs_color_dist) and (color in mock_color_dist)):
                 continue
-
-            obinctr = self.obs_color_dist[color]['binctr']
-            mbinctr = mock_color_dist[color]['binctr']
-            opdf_smooth = self.obs_color_dist[color]['pdf_smooth']
-            mpdf_smooth = mock_color_dist[color]['pdf_smooth']
-            ocdf = self.obs_color_dist[color]['cdf']
-            mcdf = mock_color_dist[color]['cdf']
-
-            xmin = np.min([mbinctr[find_first_true(mcdf > 0.001)],
+            if color in mock_color_dist:
```

Comment:
Lines 226 to 228, simplify the logic here:
```python
if color not in mock_color_dist or (self.validation_catalog and color not in self.obs_color_dist):
    continue
```
Unindent 4 spaces for Lines 229 to 281.